discovery-provider/plugins/pedalboard/apps/trending-challenge-rewards/src/app.ts

```diff
@@ -0,0 +1,279 @@
+import App from "basekit/src/app";
+import { Knex } from "knex";
+import { Ok, Err, Result } from "ts-results";
+import { AudiusLibs } from "@audius/sdk";
+import { SharedData } from "./config";
+import {
+  ChallengeDisbursementUserbank,
+  ChallengeDisbursementUserbankFriendly,
+  getChallengesDisbursementsUserbanks,
+  getChallengesDisbursementsUserbanksFriendly,
+  getTrendingChallenges,
+} from "./queries";
+import fetch from "node-fetch";
+import axios from "axios";
+import { RespondFn, SlashCommand } from "@slack/bolt";
+import { WebClient } from "@slack/web-api";
+import { formatDisbursementTable } from "./slack";
+
+// TODO: move something like this into App so results are commonplace for handlers
+export const onCondition = async (app: App<SharedData>): Promise<void> => {
+  // const { dryRun } = app.viewAppData()
+  // const disburse = await onDisburse(app, dryRun);
+  // disburse.mapErr(console.error);
+};
+
+// dryRun separated out so it can be run manually via another trigger
+export const onDisburse = async (
+  app: App<SharedData>,
+  dryRun: boolean,
+  slack: { command: SlashCommand; client: WebClient }
+): Promise<Result<undefined, string>> => {
+  const db = app.getDnDb();
+  const libs = app.viewAppData().libs;
+  const { command, client } = slack;
+
+  const startingBlockRes = await findStartingBlock(db);
+  if (startingBlockRes.err) return startingBlockRes;
+  const [startingBlock, specifier] = startingBlockRes.unwrap();
+
+  const formattedResults = formatDisbursementTable(await getChallengesDisbursementsUserbanksFriendly(db, specifier))
+  console.log(formattedResults)
+
+  await client.chat.postMessage({
```

Comment:
Let's move posting to slack after the getAllChallenges so we only post after it works?